src/StoreApi/Routes/Checkout.php

```diff
@@ -147,6 +147,14 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 		$order_controller = new OrderController();
 		$order_object     = $this->get_draft_order_object( $this->get_draft_order_id() );
 
+		// In some cases, the order might not be created yet, that might happen
+		// when trying to checkout directly from the Cart block with an express
+		// payment method. In that case, try creating a draft order first.
+		if ( ! $order_object instanceof \WC_Order ) {
+			$order_object = $this->create_or_update_draft_order();
+		}
```

Comment:
I think this is okay but I wonder if we should just modify `get_draft_order_object` so that it returns a new one if there is no draft order id provided. From the doc blocks on the method, it seems like that's what it was intended to do. So basically in `get_draft_order_object` there'd be this:

```php
// old line
// $draft_order_object = $order_id ? wc_get_order( $order_id ) : false;
$draft_order_object = $order_id ? wc_get_order( $order_id ) : $this->create_or_update_draft_order();
```

Reply:
In https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3403/commits/2577e4be8c3b41278b008e40f6816cc69078b5bf I applied a small refactor in line with your suggestion: making `get_draft_order_object()` return a new order object if no draft order existed. I needed to refactor `create_or_update_draft_order()` so it didn't depend on `get_draft_order_object()`. Also, I couldn't apply your suggestion directly because there is no guarantee the order ID will be of a draft order: the order ID stored in `store_api_draft_order` might be pointing to an order that is no longer a draft if the user has made a previous purchase.